### DISCUSSION TITLE
Using BPF to ignore blocks of IPs

### DISCUSSION THREAD

**Discussion by JoshTickles** [Security-Onion-Solutions/securityonion#4604] [Unsupported Versions]
Hey all,

I'm having some trouble using BPF within my distributed deployment.
1 x sensor
1 x master
2 x search
I am running the lastest deployment of SO. 

I'd like some guidance / sanity checking that my approach for BPF is correct. 

Short version - We have tenable cloud that performs web app scans on some of our external endpoints. I want to add these IP ranges to a BPF for Security Onion to ignore anything from them.

Here is a snippit of the section I am adding to my `/opt/so/saltstack/local/pillar/global.sls` file after following the docs here - https://docs.securityonion.net/en/latest/bpf.html?highlight=BPF

```
nids:
 bpf:
# Start - Nessus / Tenable IP Ranges
# https://docs.tenable.com/tenableio/vulnerabilitymanagement/Content/Scans/Classic/Scanners.htm
# 
# Amazon CA-CENTRAL-1 (Canada)
  - not src net 35.182.14.64/26
  - not src net 3.98.92.0/25
# Amazon US-EAST (Ohio or Virginia)
  - not src net 13.59.252.0/25
  - not src net 54.175.125.192/26
  - not src net 34.201.223.128/25
  - not src net 3.132.217.0/25
  - not src net 18.116.198.0/24
  - not src net 44.192.244.0/24
# Amazon US-WEST (California or Oregon)
  - not src net 54.219.188.128/26
  - not src net 13.56.21.128/25
  - not src net 34.223.64.0/25

```

I have tried adding this under `nids` and also under `steno`. I'd prefer to use steno if possible, just to keep things cleaner. 
I have waited the 15 mins it takes to sync salt, and I've also tried `sudo salt '*' state.highstate`. 
Regardless of what I try, the traffic is still captured, and alerts still show in my SoC console. 

Any advice?

Cheers, Josh.

**Comment by Masaya-A** [comment]
https://docs.securityonion.net/en/latest/bpf.html#multiple-conditions
> If your BPF contains multiple conditions you can put them on multiple lines and join them with `&&` but make sure the final condition has no `&&` at the end.

**Reply by JoshTickles** [reply]
Should have specified that I have tried this. I get an error via this method when running salt highstate. 

```----------
          ID: bpfcompilationfailure
    Function: test.configurable_test_state
      Result: False
     Comment: BPF Compilation Failed - Discarding Specified BPF
     Started: 21:18:12.605749
    Duration: 2.665 ms
     Changes: ```

**Reply by Masaya-A** [reply]
How about Suricata? (I think that error is from so-pcap = Steno)
Does another `steno:` section already exist in your `global.sls`?

**Reply by JoshTickles** [reply]
Thanks for checking.

No other `steno` in the config.
Using `nids` results in:

```----------
          ID: suribpfcompilationfailure
    Function: test.configurable_test_state
      Result: False
     Comment: BPF Syntax Error - Discarding Specified BPF
     Started: 01:12:04.467558
    Duration: 2.808 ms
     Changes:   
----------
```

**Reply by Masaya-A** [reply]
Well, I recommend you should try a simple rule like this:
```
nids:
 bpf:
  - not src net 35.182.14.64/26
steno:
 bpf:
  - not src net 35.182.14.64/26
```
If success, then:
```
nids:
 bpf:
  - not src net 35.182.14.64/26 &&
  - not src net 3.98.92.0/25
steno:
 bpf:
  - not src net 35.182.14.64/26 &&
  - not src net 3.98.92.0/25
```

**Comment by manuelplascencia** [comment]
Should quotes be used ? I see that sometimes?

Thank you,
Manuel Plascencia
Director of Technology & Information Services
Whittier City School District

(562)789-3261
***@***.***
Follow us @WhittierCitySD on FB,  
and @WhittierCity on Twitter
San Gabriel Valley Tech Exchange - (Subscribe)
CITE Region Chair

**Comment by JoshTickles** [comment]
I've managed to solve my issue. 

Note - Be careful with spacing and things in the right place! All sorted now. Thanks for your help @Masaya-A @manuelplascencia .

**Comment by JoshTickles** [comment]
Actually jumped the gun here - Thought it had worked (I get not salt errors now), but Steno it still capturing packets for these IPs.... :(
